lxd/api_metrics.go

```diff
@@ -378,10 +379,40 @@ func getFilteredMetrics(s *state.State, r *http.Request, compress bool, metricSe
 	return response.SyncResponsePlain(true, compress, metricSet.String())
 }
 
-func internalMetrics(ctx context.Context, daemonStartTime time.Time, tx *db.ClusterTx) *metrics.MetricSet {
```

Comment:
the change to this function's signature should be a separate commit as not related to the commit message

Reply:
Personally I would have plumbed in the replacement of daemonStartTime argument with *state.State as single commit first, including passing s.StartTime instead of daemonStartTime to `out.AddSamples()`.

Then as another commit I would have introcuced the `clusterMemberWarnings` function and replaced the call to ` dbCluster.GetWarnings` in `internalMetrics`.

As it is now the changes are munged together and harder to review.

Reply:
Noted! I find it useful to understand how you think about this so I can try to make your life easier :)

I will be pushing these changes along with the tests